ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -313,6 +307,21 @@ private[commands] class CommandTracker[Context](maxDeduplicationTime: () => JDur
     logic -> promise.future
   }
 
+  private def maxTimeoutFromDeduplicationPeriod(deduplicationPeriod: DeduplicationPeriod) = {
+    lazy val maxDeduplicationDuration = maxDeduplicationTime()
+    val timeoutDuration = deduplicationPeriod match {
+      case DeduplicationPeriod.Empty =>
+        maxDeduplicationDuration
+      case DeduplicationPeriod.DeduplicationTime(duration) =>
+        JDuration.ofSeconds(duration.seconds, duration.nanos.toLong)
+      case DeduplicationPeriod.DeduplicationStart(_) =>
+        maxDeduplicationDuration
```

Comment:
Shouldn't this raise an error if the deduplication start is earlier than what the `maxDeduplciationDuration` allows?

Reply:
That should already be the case, as by this point the deduplication should've already been validated. To be true to specifics though I will calculate the deduplication duration here and not just use the max duration.